### DISCUSSION TITLE
Accessibility Improvement - Cursor Locator

### DISCUSSION THREAD

**Discussion by jquesada2016** [helix-editor/helix#10210] [Ideas]
An issue which is very often overlooked for visually impaired users is a cursor locating strategy. Ideally, I would hope this would be something possible when the eventual plugin system is rolled out for maximum customizability.

This is just a mechanism for quickly locating where the cursor is on the screen. The easiest way to achieve this is to draw an n-by-n crosshair or rectangle around the cursor. This can be persistent, toggled, or only shown while holding down some special key, such as CTRL.

there are many other possible strategies too, such as hiding all text on the screen except for the current line, etc.

**Comment by jquesada2016** [comment]
I'll check out the source code as soon as I get a chance to see if I can get something like this up and running.

**Comment by the-mikedavis** [comment]
Cursorline was merged recently which might provide the horizontal component of this: https://github.com/helix-editor/helix/pull/2170

**Comment by AceofSpades5757** [comment]
@jquesada2016 I've never heard of a cross-hair or box around the cursor, but that sounds _interesting_. Below are the settings to the most common solution I've seen, commonly named `cursorline`.

<u>**This is on the latest, `master` branch of Helix. These features are currently not supported in the most recent stable release.**</u>

Here's the setting to enable it.

```toml
[editor]
cursorline = true
```

https://docs.helix-editor.com/master/configuration.html?highlight=cursorline#editor-section

---

And here are the settings in the theme documentation (you would need to create your own or edit an existing).

https://docs.helix-editor.com/master/themes.html?highlight=cursorline#Interface

**Comment by jquesada2016** [comment]
@AceofSpades5757 Yeah, not in a terminal editor. It's something that has always stopped me from using vim in the past. The cursor jumps around and I always loose it. It is very common, however, for accessible software aids targeting the visually impaired.
